### PR TITLE
feat: Allow registering absolute URLs for autorefs

### PR DESCRIPTION
**Pull request by oprypin**
For now this is not used for anything, but the refactor is good for whatever plan we decide to go through regarding inventories.

### CODE REVIEW

File: src/mkdocs_autorefs/plugin.py [hunk 1]
```diff
@@ -69,13 +84,22 @@ def get_item_url(self, anchor: str) -> str:
...
-            new_anchor = self.get_fallback_anchor(anchor)
-            if new_anchor and new_anchor in self._url_map:
-                return self._url_map[new_anchor]
+            if identifier in self._abs_url_map:
+                return self._abs_url_map[identifier]
+
+            if fallback:
+                new_identifier = fallback(identifier)
+                if new_identifier:
+                    return self.get_item_url(new_identifier, from_url)
```
Comment (pawamoy) [inline src/mkdocs_autorefs/plugin.py:95]: Don't we risk infinite recursion here?

**Review by pawamoy — COMMENTED** [review]
Just one comment.

File: src/mkdocs_autorefs/plugin.py (hunk 1 above)
Comment (oprypin) [reply src/mkdocs_autorefs/plugin.py:95]: Oh, sorry that there's no explanation for the "cleverness".

The self-call is not exactly the same, `fallback` is dropped. So, no, there will be only 2 attempts.

I'll add an explicit `fallback=None` for clarity.

Comment (pawamoy) [reply src/mkdocs_autorefs/plugin.py]: Ah, my bad for not reading this more thoroughly, I would have seen it :slightly_smiling_face: 
Thanks for the explanation. The `fallback=None` is not necessary, but you can add it if you want.

Comment (pawamoy) [reply src/mkdocs_autorefs/plugin.py]: And you did. Thanks again ^^

Comment (oprypin) [reply src/mkdocs_autorefs/plugin.py]: Ooh let me revert that, there's some failure, and it's a concern for backwards compatibility.
I'll also check in detail why the test failed, I would not have expected that

Comment (oprypin) [reply src/mkdocs_autorefs/plugin.py:95]: This is resolved btw. There wasn't really a problem, just an implicit merge conflict

**Pull request merged**